src/group_theory/perm/basic.lean

```diff
@@ -30,7 +30,15 @@ instance perm_group : group (perm α) :=
   mul_assoc := λ f g h, (trans_assoc _ _ _).symm,
   one_mul := trans_refl,
   mul_one := refl_trans,
-  mul_left_inv := self_trans_symm }
+  mul_left_inv := self_trans_symm,
+  npow := λ n f, ⟨f^[n], f.symm^[n], f.left_inv.iterate _, f.right_inv.iterate _⟩,
```

Comment:
If you provide this as a `has_pow` instance separately, then you don't need to repeat this definition in zpow.

Reply:
Wrong. `zpow_rec` completely ignores existing `pow` instances, which is why I opened #17826.

Reply:
What I meant by that is that you could change
```diff
- | int.of_nat n := λ f, ⟨f^[n], f.symm^[n], f.left_inv.iterate _, f.right_inv.iterate _⟩
+ | int.of_nat n := λ f, f^n
```

Reply:
Oh yeah. I don't think that's worth optimising.